m4/cc_retpoline.m4

```diff
@@ -0,0 +1,21 @@
+dnl
+dnl Check for support for Retpoline
+dnl
+
+AC_DEFUN([AC_CC_RETPOLINE],[
+    AC_REQUIRE([gl_UNKNOWN_WARNINGS_ARE_ERRORS])
+    if test $enable_hardening = yes; then
+      case "$host" in
+        *)
+          gl_COMPILER_OPTION_IF([-mfunction-return=thunk -mindirect-branch=thunk], [
+            CFLAGS="$CFLAGS -mfunction-return=thunk -mindirect-branch=thunk"
+            ],
+            [],
+            [AC_LANG_PROGRAM([[
+  #include <pthread.h>
+  __thread unsigned int t_id;
```

Comment:
Why is the test program using pthread?

Reply:
Hello,

We have based ourselves on existing implementations.

What would you recommend in the Dovecot context?

Reply:
Oh, I see the same pthread code is used in several other m4 checks already in Dovecot. I wonder why. Dovecot generally doesn't use pthreads, so I was just thinking that pthread usage in these checks is a bit confusing, as if it's important to use them only with pthreads.

Reply:
That's just how they are usually written, it's not about whether pthreads are used or not.

Reply:
It seems empty body is fine too, so `AC_LANG_PROGRAM()`. It generates into `return 0;` only.